Sources/ExistentialFuture.swift

```diff
@@ -61,6 +61,17 @@ private final class FilledFuture<Value>: FutureBase<Value> {
     }
 }
 
+// Concrete future wrapper that will never get filled.
+private final class UnfilledFuture<Value>: FutureBase<Value> {
```

Comment:
What do you think about calling this `Never<Value>`? `UnfilledFuture` reads to me like the door is open for it to be a view onto a `Future` that might one day be filled, and just that that wouldn't be visible through the `UnfilledFuture` view. `Never<Value>` seems straight-up "yeah, this'll never happen". As a bonus, it's a simple word rather than a compound. :)